pkgs/os-specific/linux/minimal-bootstrap/coreutils/static.nix

```diff
@@ -0,0 +1,74 @@
+{ lib
+, buildPlatform
+, hostPlatform
+, fetchurl
+, bash
+, gcc
+, musl
+, binutils
+, gnumake
+, gnused
+, gnugrep
+, gawk
+, diffutils
+, findutils
+, gnutar
+, gzip
+}:
+let
+  inherit (import ./common.nix { inherit lib; }) meta;
+  pname = "coreutils-static";
+  version = "9.4";
+
+  src = fetchurl {
+    url = "mirror://gnu/coreutils/coreutils-${version}.tar.gz";
+    hash = "sha256-X2ANkJOXOwr+JTk9m8GMRPIjJlf0yg2V6jHHAutmtzk=";
+  };
```

Comment:
Can this be deduped with the musl variant?

Reply:
There's a motivation to prevent dependency cycles and rebuild loops, and favouring simpler code over deduplication. Important to the "trust" factor will be the the ability for bootstrap to be readable and understandable by end users who wish to, and duplicating code so as to not need to jump between files helps with that I think.

Hope this explanation makes sense? It's something that I need to write up as part of the documentation deliverable for the STF fund.

Reply:
> There's a motivation to prevent dependency cycles and rebuild loops, and favouring simpler code over deduplication.

Per what I wrote in https://github.com/NixOS/nixpkgs/pull/260193#discussion_r1367978807, I rather have duplication than simpler code.

My guess is that code *I* wrote was getting within @emilytrau's way, which is understandable. So I think it would be good if newer deduplication we had was driven by @emilytrau so the burden of untangling on first readthrough will be on the rest of us, not her.

In previous PRs I was saying this can wait to the end, but now I that we are getting nearer a combinatorial explosion of stuff, I am getting wary about continuing to wait.

Reply:
FWIW I think we can do better than the rest of Nixpkgs by preferring Nix-language-level than shell-level abstractions. The benefit is then the final produced code that is in the derivation (try `nix derivation show | jq ...`) will have all those abstractions removed, and what is left is simple (but verbose) commands! Best of both worlds!